students/JaeKim/lesson04/assignment/tests/test_basic_operations.py

```diff
@@ -0,0 +1,202 @@
+        add_customer(**user)
+
+    active_customers = list_active_customers()
+    assert active_customers[0]['first_name'].lower() == 'jae'
+    assert active_customers[1]['first_name'].lower() == 'erin'
+    assert active_customers[2]['first_name'].lower() == 'lebron'
+
+
+def test_add_ok_customer(setup_test_db):
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    add_customer(**OK_CUSTOMER)
+    valid_customer = Customer.get(Customer.customer_id == OK_CUSTOMER['customer_id'])
+
+    assert valid_customer.first_name == OK_CUSTOMER['name']
+    assert valid_customer.customer_id == OK_CUSTOMER['customer_id']
+    assert valid_customer.last_name == OK_CUSTOMER['lastname']
+    assert valid_customer.home_address == OK_CUSTOMER['home_address']
+    assert valid_customer.phone_number == OK_CUSTOMER['phone_number']
+    assert valid_customer.email_address == OK_CUSTOMER['email_address']
+    assert valid_customer.status == OK_CUSTOMER['status']
+    assert valid_customer.credit_limit == OK_CUSTOMER['credit_limit']
+
+
+def test_credit_limit_float(setup_test_db):
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    bad_customer = dict(OK_CUSTOMER)
+    bad_customer['credit_limit'] = '$40'
+
+    with pytest.raises(ValueError):
+        add_customer(**bad_customer)
+
+
+def test_get_customer(setup_test_db):
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    add_customer(**OK_CUSTOMER)
+    valid_customer = search_customer(OK_CUSTOMER['customer_id'])
+
+    assert valid_customer.first_name == OK_CUSTOMER['name']
+    assert valid_customer.customer_id == OK_CUSTOMER['customer_id']
+    assert valid_customer.last_name == OK_CUSTOMER['lastname']
+    assert valid_customer.home_address == OK_CUSTOMER['home_address']
+    assert valid_customer.phone_number == OK_CUSTOMER['phone_number']
+    assert valid_customer.email_address == OK_CUSTOMER['email_address']
+    assert valid_customer.status == OK_CUSTOMER['status']
+    assert valid_customer.credit_limit == OK_CUSTOMER['credit_limit']
+#
+
+def test_delete_customer(setup_test_db):
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    # Add and assert to confirm record exists before deleting
+    add_customer(**OK_CUSTOMER)
+    valid_customer = search_customer(OK_CUSTOMER['customer_id'])
+
+    assert valid_customer.first_name == OK_CUSTOMER['name']
+    assert valid_customer.customer_id == OK_CUSTOMER['customer_id']
+    assert valid_customer.last_name == OK_CUSTOMER['lastname']
+    assert valid_customer.home_address == OK_CUSTOMER['home_address']
+    assert valid_customer.phone_number == OK_CUSTOMER['phone_number']
+    assert valid_customer.email_address == OK_CUSTOMER['email_address']
+    assert valid_customer.status == OK_CUSTOMER['status']
+    assert valid_customer.credit_limit == OK_CUSTOMER['credit_limit']
```

Comment:
If you are asserting a delete, you can assume the customer is there, since you tested adding above. Just add, and then make sure you can delete.